### ISSUE TITLE
cleanup leaked ENIs race

### ISSUE THREAD

**Issue by nithu0115** [aws/amazon-vpc-cni-k8s#1099]
While creating new ENIs on a worker node, if the cleanup leaked ENIs go-routine kicks in, and if the status of the ENI is available when describe network is called, the cleanup leaked ENI routine will delete the ENI. 

Current workflow: Worker node A CreateNetworkInterface API call that create ENI-1234 -> later calls CreateTags API on the newly created ENI. At this point in time, worker node B make a DescribeNetworkInterfaces API call as part of cleanup leaked ENIs routine which checks every 1 hour for dangling/leaked ENI with tag filters 
```
"filterSet": {
            "items": [
                {
                    "name": "tag-key",
                    "valueSet": {
                        "items": [
                            {
                                "value": "node.k8s.amazonaws.com/instance_id"
                            }
                        ]
                    }
                },
                {
                    "name": "status",
                    "valueSet": {
                        "items": [
                            {
                                "value": "available"
                            }
                        ]
                    }
                }
            ]
        }
```
As soon as cleanup go-routine finds ENI in available state, i will make go ahead and call DeleteNetworkInterface which in the case is wrong problems on worker node A.

To mitigate this issue, I am planning to introduce a new tag called creating and remove it as soon as ENI goes to `in-use` status. Thoughts ?

CC: @mogren @jayanthvn @anguslees

**Comment by mogren** [comment]
The current implementation uses `getFilteredListOfNetworkInterfaces() that has the following filter when describing ENIs in the VPC:
```
	// The tag key has to be "node.k8s.amazonaws.com/instance_id"
	tagFilter := &ec2.Filter{
		Name: aws.String("tag-key"),
		Values: []*string{
			aws.String(eniNodeTagKey),
		},
	}
	// Only fetch "available" ENIs.
	statusFilter := &ec2.Filter{
		Name: aws.String("status"),
		Values: []*string{
			aws.String("available"),
		},
	}
```
The tag `node.k8s.amazonaws.com/instance_id` is added in `tagENI()` only after the ENI has been attached, so if the ENI is available and doesn't have that tag, we assumed it is in the state where it has been created, but not yet attached to the node it was created for. When testing this, I did not see the loop deleting untagged ENIs.

That said, with the recent addition of the "Tag on Create" API, we could tag these ENIs with "CreatedAt" and a timestamp, so that we only delete ENIs that have been stuck in  "Available" for at least 15 minutes or something like that. To delete the tag, we would have to add that Permission as well to the CNI policy and make an additional EC2 call.

**Comment by mogren** [comment]
Fixed in #1109
